src/Spectre.libClassifier/OpenCvDataset.cpp

```
@@ -52,14 +65,15 @@ OpenCvDataset::OpenCvDataset(gsl::span<const DataType> data, gsl::span<const Lab
     }
     const auto numberOfColumns = data.size() / labels.size();
     auto rowBegin = m_Data.data();
-    for (auto i=0u; i < labels.size(); ++i)
+    for (auto i = 0u; i < labels.size(); ++i)
     {
         m_observations[i] = Observation(rowBegin, numberOfColumns);
         rowBegin += numberOfColumns;
     }
 }
 
 OpenCvDataset::OpenCvDataset(cv::Mat data, cv::Mat labels):
+    m_isEmpty(data.rows * data.cols),
```

Here we are silently converting from size to `bool`.